hack/shoot-comparator/pkg/shoot/matcher_test.go

```diff
@@ -107,5 +134,986 @@ var _ = Describe(":: shoot matcher :: ", func() {
+				Region: "test1",
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				DNS: &v1beta1.DNS{
+					Domain: ptr.To[string]("test2"),
+				},
+				Region: "test1",
+			})),
+			false,
+		),
+		Entry(
+			"should detect differences in spec/dns/providers",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				DNS: &v1beta1.DNS{
+					Domain: ptr.To[string]("test"),
+					Providers: []v1beta1.DNSProvider{
+						{
+							Type:       ptr.To[string]("test1"),
+							Primary:    ptr.To[bool](true),
+							SecretName: ptr.To[string]("test"),
+						},
+					},
+				},
+				Region: "test1",
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				DNS: &v1beta1.DNS{
+					Domain: ptr.To[string]("test"),
+				},
+				Region: "test1",
+			})),
+			false,
+		),
+		Entry(
+			"should find no differences in spec/dns/providers",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				DNS: &v1beta1.DNS{
+					Domain: ptr.To[string]("test"),
+					Providers: []v1beta1.DNSProvider{
+						{
+							Type:       ptr.To[string]("test1"),
+							Primary:    ptr.To[bool](true),
+							SecretName: ptr.To[string]("test"),
+						},
+					},
+				},
+				Region: "test1",
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				DNS: &v1beta1.DNS{
+					Domain: ptr.To[string]("test"),
+					Providers: []v1beta1.DNSProvider{
+						{
+							Type:       ptr.To[string]("test1"),
+							Primary:    ptr.To[bool](true),
+							SecretName: ptr.To[string]("test"),
+						},
+					},
+				},
+				Region: "test1",
+			})),
+			true,
+		),
+		Entry(
+			"should find differences in spec/tolerantions #1",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key:   "key",
+						Value: ptr.To[string]("val"),
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{})),
+			false,
+		),
+		Entry(
+			"should find differences in spec/tolerantions #2",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+					{
+						Key: "key2",
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			false,
+		),
+		Entry(
+			"should find no differences in spec/tolerantions",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should skip extra items in spec/tolerantions",
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{})),
+			deepCp(empty, withShootSpec(v1beta1.ShootSpec{
+				Tolerations: []v1beta1.Toleration{
+					{
+						Key: "key",
+					},
+				},
+			})),
+			true,
+		),
+		Entry(
+			"should find differences in spec/maintenance #1",
```

Comment:
The test doesn't cover fields from `maintenance` that are ignored (`timeWindow`, `confineSpecUpdateRollout`).

Reply:
Will be cover [here](https://github.com/kyma-project/infrastructure-manager/blob/e0c9da5f2fbed9880b4c8f8c9d48e7fcf430a3fe/hack/shoot-comparator/pkg/shoot/matcher_test.go#L329).